lib/chef/resource/windows_task.rb

```diff
@@ -163,27 +185,44 @@ def validate_create_frequency_modifier(frequency, frequency_modifier)
               raise ArgumentError, "frequency_modifier value #{frequency_modifier} is invalid. Valid values for :weekly frequency are 1 - 52."
             end
           when :monthly
-            unless ("1".."12").to_a.push("FIRST", "SECOND", "THIRD", "FOURTH", "LAST", "LASTDAY").include?(frequency_modifier.to_s.upcase)
+            unless ("1".."12").include?(frequency_modifier.to_s.upcase) || frequency_modifier_includes_days_of_weeks?(frequency_modifier)
               raise ArgumentError, "frequency_modifier value #{frequency_modifier} is invalid. Valid values for :monthly frequency are 1 - 12, 'FIRST', 'SECOND', 'THIRD', 'FOURTH', 'LAST', 'LASTDAY'."
             end
           end
         end
       end
 
-      def validate_create_day(day, frequency)
+      def validate_create_day(day, frequency, frequency_modifier)
         unless [:weekly, :monthly].include?(frequency)
           raise "day property is only valid for tasks that run monthly or weekly"
         end
         if day.is_a?(String) && day.to_i.to_s != day
           days = day.split(",")
-          days.each do |d|
-            unless ["mon", "tue", "wed", "thu", "fri", "sat", "sun", "*"].include?(d.strip.downcase)
-              raise ArgumentError, "day property invalid. Only valid values are: MON, TUE, WED, THU, FRI, SAT, SUN and *. Multiple values must be separated by a comma."
+          if days_includes_days_of_months?(days)
+            if frequency == :weekly
+              raise "day values 1-31 or last is only valid with frequency :monthly"
+            end
+
+            if frequency_modifier && frequency_modifier != 1
+              raise "Do not use frequency_modifier property when day values set as 1-31 or last"
+            end
+          else
+            days.each do |d|
+              unless ["mon", "tue", "wed", "thu", "fri", "sat", "sun", "*"].include?(d.strip.downcase)
+                raise ArgumentError, "day property invalid. Only valid values are: MON, TUE, WED, THU, FRI, SAT, SUN and *. Multiple values must be separated by a comma."
+              end
             end
           end
         end
       end
 
+      def days_includes_days_of_months?(days)
+        days.each do |d|
```

Comment:
This method is very costly because if days have 5 element the below array [1..31, last] create 5 times. 

This method should be.
```
def days_includes_days_of_months?(days)
    days.all? { |day| default_array.include?(day) }
end

def default_array
   ("1".."31").to_a << "last"
end
```

Reply:
I don't think that the days.all? { |d| } block is reducing the cost since that is also enumeration it may increase the readability.

Reply:
I used array set operation instead.

Reply:
That's make sense, you can change `days.all? { |day| valid_days.include?(day) }` to `(days - valid_days).empty?`